<launch>

  <!-- 1) Gazebo Arguments -->
  <arg name="world_name"   default="$(find tracker_control)/worlds/test1.world"/>
  <arg name="paused"       default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui"          default="true"/>
  <arg name="headless"     default="false"/>
  <arg name="debug"        default="false"/>

  <!-- 2) MSCKF Print Level -->
  <arg name="verbosity" default="ALL"/> <!-- ALL, DEBUG, INFO, WARNING, ERROR, SILENT -->

  <!-- Start Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name"   value="$(arg world_name)"/>
    <arg name="paused"       value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui"          value="$(arg gui)"/>
    <arg name="headless"     value="$(arg headless)"/>
    <arg name="debug"        value="$(arg debug)"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro '$(find tracker_control)/models/sliping_robot_motor/urdf/sliping_robot_motor.xacro'" />

  <!-- Step 4: Spawn the robot into Gazebo -->
  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" output="screen"
        args="-urdf -param robot_description -model sliping_robot_motor -x 4 -y 0 -z 0.3" />



  <node name="fl_controller_spawner" pkg="controller_manager"
        type="spawner" args="fl_motor_control" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_map" output="screen"
        args="-4 0 0 0 0 0 world odom"/>

  <arg name="config_path" default="$(find tracker_control)/config/estimator_config.yaml" />

  <!-- MASTER NODE! -->
  <node name="tracker_control"
        pkg="tracker_control"
        type="image_analyzer_ekf.py"
        output="screen">
  </node>

  <node pkg="image_view" type="image_publisher" name="image_pub" output="screen">
    <param name="filename" value="../lab.jpg" />
    <remap from="image" to="/camera/image_raw" />
  </node>

</launch>
